### PR TITLE
Fix bug with demand decreasing for non pulled Components

### PR DESCRIPTION
**Pull request by codeadict**
**Description**

Fixes issue with demand going down for non pulled components. 

**Motivation and Context**

The major part of this fix is on https://github.com/system76/assembly/pull/13. This PR addresses 2 major issues:

* Assembly GRPC client was not connecting correctly after upgrading the Elixir https://github.com/elixir-grpc/grpc library.
* The Insert monitor created recently caused race conditions between the warmup of components and data that starts coming from Assembly, causing components demand to be reset to 0 in memory. This was removed and will mean we will need to restart warehouse again to pick up new components, component updates are still handled well. This will be addressed in a follow-up ticket to detect new components in HAL and send the information here without the periodically reading the DB trickery.

**Related Issues**

* https://github.com/system76/assembly/pull/13

### CODE REVIEW

File: test/warehouse/sku_test.exs [hunk 1]
```diff
@@ -16,6 +16,14 @@ defmodule Warehouse.SkuTest do
     assert Sku.list_skus(ids) == skus
   end
 
+  test "warmup_skus/0 starts all sku supervisors" do
+    sku = insert(:sku)
+    assert Warehouse.SkuRegistry |> Registry.lookup(to_string(sku.id)) |> length() == 0
+
+    Sku.warmup_skus()
+    assert Warehouse.SkuRegistry |> Registry.lookup(to_string(sku.id)) |> length() == 1
```
Comment (macifell) [inline test/warehouse/sku_test.exs:24]: :+1:

File: test/warehouse/component_test.exs [hunk 2]
```diff
@@ -28,6 +29,14 @@ defmodule Warehouse.ComponentTest do
     assert Component.list_components(ids) == components
   end
 
+  test "warmup_components/0 starts all component supervisors" do
+    component = insert(:component)
+    assert Warehouse.ComponentRegistry |> Registry.lookup(to_string(component.id)) |> length() == 0
+
+    Component.warmup_components()
+    assert Warehouse.ComponentRegistry |> Registry.lookup(to_string(component.id)) |> length() == 1
```
Comment (macifell) [inline test/warehouse/component_test.exs:37]: :+1:

File: lib/warehouse/clients/assembly/connection.ex [hunk 3]
```diff
@@ -43,7 +43,7 @@ defmodule Warehouse.Clients.Assembly.Connection do
 
   @impl true
   def handle_info({:gun_up, _, _, _, _}, _state) do
-    Logger.debug("Warehouse.Clients.Assembly.Connection connected")
+    Logger.info("Warehouse.Clients.Assembly.Connection connected")
```
Comment (macifell) [inline lib/warehouse/clients/assembly/connection.ex:46]: Good idea to always display these messages.

File: lib/warehouse/application.ex [hunk 4]
```diff
@@ -30,6 +29,14 @@ defmodule Warehouse.Application do
 
     opts = [strategy: :one_for_one, name: Warehouse.Supervisor]
 
-    {:ok, _pid} = Supervisor.start_link(children, opts)
+    with {:ok, pid} <- Supervisor.start_link(children, opts) do
+      Task.Supervisor.async_nolink(Warehouse.TaskSupervisor, fn ->
+        :warehouse
+        |> Application.get_env(:warmup)
+        |> apply([])
```
Comment (macifell) [inline lib/warehouse/application.ex:36]: Makes sense

**Review by macifell — APPROVED** [review]
Looks good to me!

File: lib/warehouse/clients/assembly/connection.ex (hunk 3 above)
Comment (codeadict) [reply lib/warehouse/clients/assembly/connection.ex:46]: Yes for now, until we can change the log level live

**Pull request merged**